contrib/olivere/elastic/elastictrace.go

```diff
@@ -115,7 +118,22 @@ func peek(rc io.ReadCloser, max int, n int) (string, io.ReadCloser, error) {
 	}
 	snip, err := r.Peek(n)
 	if err == io.EOF {
-		err = nil
+		return string(snip), rc2, nil
```

Comment:
This is not what I meant. If you do this, you will never get to unpack response bodies that are shorter than `n` and are gzipped. What I meant was:
```go
if err == io.EOF {
    err = nil
}
if err != nil {
    return string(snip), rc2, err
}
```

Reply:
Sorry for the misunderstand. Thanks for clarifying.